plugins/lookup/onepassword_ssh_key.py

```diff
@@ -0,0 +1,148 @@
+      _terms:
+        description: Identifier(s) (case-insensitive UUID or name) of item(s) to retrieve.
+        required: true
+        type: list
+        elements: string
+      ssh_format:
+        description: Output key in SSH format if true. Otherwise, outputs in the default format.
+        required: false
+        default: false
+        type: bool
+
+    extends_documentation_fragment:
+      - community.general.onepassword
+      - community.general.onepassword.lookup
+"""
+
+EXAMPLES = """
+- name: Retrieve the private key of ssh key from 1Password
+  ansible.builtin.debug:
+    var: lookup('community.general.onepassword_ssh_key', 'SSH Key', ssh_format=true)
+"""
+
+RETURN = """
+  _raw:
+    description: Private key of SSH key
+    type: list
+    elements: string
+"""
+import json
+
+from ansible_collections.community.general.plugins.lookup.onepassword import (
+    OnePass,
+    OnePassCLIv2,
+)
+from ansible.errors import AnsibleLookupError
+from ansible.module_utils.common.text.converters import to_bytes
+from ansible.plugins.lookup import LookupBase
+
+
+class OnePassCLIv2SSHKey(OnePassCLIv2):
+
+    def _get_raw(self, item_id, vault=None, token=None):
+        args = ["item", "get", item_id, "--format", "json"]
+        if vault is not None:
+            args = [*args, f"--vault={vault}"]
+
+        if self.service_account_token:
+            if vault is None:
+                raise AnsibleLookupError(
+                    "'vault' is required with 'service_account_token'"
+                )
+
+            environment_update = {
+                "OP_SERVICE_ACCOUNT_TOKEN": self.service_account_token
+            }
+            return self._run(args, environment_update=environment_update)
+
+        if token is not None:
+            args = [*args, to_bytes("--session=") + token]
+
+        return self._run(args)
+
+    def get_ssh_key(self, item_id, vault=None, token=None, ssh_format=False):
+        rc, out, err = self._get_raw(item_id, vault, token)
+
+        data = json.loads(out)
+
+        if data.get("category") != "SSH_KEY":
+            raise AnsibleLookupError(f"Item {item_id} is not SSH Key")
+
+        private_key_field = next(
+            (
+                field
+                for field in data.get("fields", {})
+                if field.get("id") == "private_key" and field.get("type") == "SSHKEY"
+            ),
+            None,
+        )
+        if not private_key_field:
+            raise AnsibleLookupError(f"No private key found for item {item_id}.")
+
+        if ssh_format:
+            return (
+                private_key_field.get("ssh_formats", {})
+                .get("openssh", {})
+                .get("value", "")
+            )
+        return private_key_field.get("value", "")
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables=None, **kwargs):
+        self.set_options(var_options=variables, direct=kwargs)
+
+        ssh_format = kwargs.get("ssh_format")
```

Comment:
```suggestion
        ssh_format = self.get_option("ssh_format")
```

Reply:
at first `self.get_option` wasn't working on my setup, but it looks okay now.